src/Histogram.ts

```diff
@@ -74,19 +84,46 @@ export default class Histogram {
     }
   }
 
-  // return the bin index of the given data value
-  public static findBin(dataValue: number, dataMin: number, binSize: number, numBins: number): number {
-    let binIndex = Math.floor((dataValue - dataMin) / binSize);
-    // for values that lie exactly on last bin we need to subtract one
-    if (binIndex === numBins) {
-      binIndex--;
+  private static findBin(
+    dataValue: number,
+    dataMin: number,
+    dataMax: number,
+    binSize: number,
+    isFloatType: boolean,
+    castToInt: boolean
+  ): number {
+    const binIndex = (dataValue - dataMin) / binSize;
+    if (!castToInt) {
+      return binIndex;
```

Comment:
related to my other comment, should this make some guarantees about out of range values?

Reply:
If we're not casting to int, we want an exact fractional bin value, even if it's out of range. I would be doing this math to make sure that the conversion from a value -> bin index -> value is stable.